LoopKit/TemporaryScheduleOverride.swift

```diff
@@ -429,7 +429,88 @@ extension TemporaryScheduleOverride.EnactTrigger: Codable {
+        apply(over: timeline) { value, override in
+            value / override.settings.effectiveInsulinNeedsScaleFactor
+        }
+    }
+
+    public func applySensitivity(over timeline: [AbsoluteScheduleValue<HKQuantity>]) -> [AbsoluteScheduleValue<HKQuantity>] {
+        apply(over: timeline) { quantity, override in
+            let value = quantity.doubleValue(for: .milligramsPerDeciliter)
+            return HKQuantity(
+                unit: .milligramsPerDeciliter,
+                doubleValue: value / override.settings.effectiveInsulinNeedsScaleFactor
+            )
+        }
+    }
+
+    public func applyBasal(over timeline: [AbsoluteScheduleValue<Double>]) -> [AbsoluteScheduleValue<Double>] {
+        apply(over: timeline) { value, override in
+            value * override.settings.effectiveInsulinNeedsScaleFactor
+        }
+    }
+
+    public func applyCarbRatio(over timeline: [AbsoluteScheduleValue<Double>]) -> [AbsoluteScheduleValue<Double>] {
+        apply(over: timeline) { value, override in
+            value * override.settings.effectiveInsulinNeedsScaleFactor
+        }
+    }
+
+    public func applyTarget(over timeline: [AbsoluteScheduleValue<ClosedRange<HKQuantity>>], at date: Date) -> [AbsoluteScheduleValue<ClosedRange<HKQuantity>>] {
+
+        guard timeline.count > 0 else {
+            return []
+        }
+
+        var applicableOverride: TemporaryScheduleOverride? = nil
+        let scheduleEndDate = timeline.last!.endDate
+
+        // Look for active or future override
+        for override in self {
+            if override.scheduledEndDate > date && override.startDate < scheduleEndDate {
+                // override is active or future
+                applicableOverride = override
+            }
+        }
+
+        if let applicableOverride, let overrideTarget = applicableOverride.settings.targetRange {
+            var result: [AbsoluteScheduleValue<ClosedRange<HKQuantity>>] = []
+
+            let overrideStart = applicableOverride.startDate
+
+            for entry in timeline {
+                if entry.startDate < overrideStart {
+                    if entry.endDate > overrideStart {
+                        result.append(
+                            AbsoluteScheduleValue(
+                                startDate: entry.startDate,
+                                endDate: overrideStart,
+                                value: entry.value
+                            )
+                        )
+                    } else {
+                        result.append(entry)
+                    }
+                    continue
+                }
+
+            }
+
+            result.append(
+                AbsoluteScheduleValue(
+                    startDate: applicableOverride.startDate,
+                    endDate: scheduleEndDate,
+                    value: overrideTarget
+                )
+            )
```

Comment:
Is it typical that the timeline does not include any elements beyond the override?

Reply:
Yes; target overrides are in effect indefinitely from the perspective of the dosing recommendation.